test/e2e/e2eutil.go

```diff
@@ -194,3 +201,80 @@ func NewTestTLSConfig(optional bool) mdbv1.TLS {
 		},
 	}
 }
+
+func ensureObject(ctx *f.Context, t *testing.T, key types.NamespacedName, obj runtime.Object) {
+	msg := ""
+	if key.Namespace != "" && key.Namespace != "default" {
+		msg = fmt.Sprintf("%s %s/%s", reflect.TypeOf(obj), key.Namespace, key.Name)
+	} else {
+		msg = fmt.Sprintf("%s %s", reflect.TypeOf(obj), key.Name)
+	}
+
+	err := f.Global.Client.Get(context.TODO(), key, obj)
+	if err != nil && !apiErrors.IsNotFound(err) {
+		t.Fatal(err)
+	} else if err == nil {
+		fmt.Printf("%s already exists!\n", msg)
+	} else {
+		err = f.Global.Client.Create(context.TODO(), obj, &f.CleanupOptions{TestContext: ctx})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+// EnsureNamespace checks that the given namespace exists and creates it if not
+func EnsureNamespace(ctx *f.Context, t *testing.T, namespace string) {
+	ensureObject(ctx, t, types.NamespacedName{Name: namespace}, &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	})
+}
+
+// EnsureServiceAccount checks that the given ServiceAccount exists and creates it if not
+func EnsureServiceAccount(ctx *f.Context, t *testing.T, namespace string, svcAcctName string) {
+	ensureObject(ctx, t, types.NamespacedName{Name: svcAcctName, Namespace: namespace}, &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      svcAcctName,
+			Namespace: namespace,
+		},
+	})
+}
+
+// EnsureRole checks that the given role exists and creates it with the given rules if not
+func EnsureRole(ctx *f.Context, t *testing.T, namespace string, roleName string, rules []rbacv1.PolicyRule) {
```

Comment:
This is not currently used, so I can delete it, but I just wanted to clarify whether we should be testing with a Role or a ClusterRole in the test created by this change first before picking which helper to delete. Thoughts?